### PR TITLE
style: add .astylerc for formatting

### PR DESCRIPTION
**Pull request by philburk**
Also add a badly formatted file for testing astyle.

Test: astyle --project=.astylerc test/style/*.c,h,cpp

### CODE REVIEW

**Comment by philburk** [comment]
See related style issues #393 and #398

**Comment by dechamps** [comment]
Drive-by comment: have you considered [`clang-format`](https://clang.llvm.org/docs/ClangFormat.html)? Not sure how it compares to `astyle` - it might behave better in the face of weird/contrived syntax, I don't know.

**Comment by philburk** [comment]
@dechamps = clang-format looks nice. It is more modern and has more options. 
I installed it on Ubuntu using:
    sudo apt-get install clang-format
Thanks for the tip.

File: .astylerc [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+# Continuation lines indented 2 levels or aligned
+--indent-continuation=2
+--max-code-length=100
+
+# Put space around operands like + and after commas.
+--pad-oper
+--pad-comma
+
+# Put space inside of parentheses
+--pad-paren-in
```
Comment (philburk) [inline .astylerc:30]: The PortAudio style is to pad the inside of parentheses, like:

    void foo( int x )

But that is an old style and not very popular. What would people think of switching to:

    void foo(int x)

**Comment by MichalPetryka** [comment]
Afaik, Visual Studio and JetBrains products (such as Resharper and CLion) support only clang-format, not astyle.

**Comment by RossBencina** [comment]
Related #239, #398

**Comment by RossBencina** [comment]
Both myself and others have tried to format PortAudio code with astyle and clang-format in the past -- and failed. I don't think either astyle nor clang-format will capture our current formatting guidelines. There are edge-cases where they clash and this will just create a mess of unnecessary and/or unwanted formatting changes.

**Pull request closed without merging**